nvme_tcp.go

```diff
@@ -317,19 +324,15 @@ func (c *NVMeTCPConnector) connectSingleDevice(ctx context.Context, info NVMeTCP
 			return Device{}, errors.New(msg)
 		}
 		if wwn == "" && len(devices) != 0 {
-			var err error
-			wwn, err = c.scsi.GetDeviceWWN(ctx, devices)
-			if err != nil {
-				logger.Debug(ctx, "wwn for devices %s not found", devices)
-			}
+			msg := "invalid WWN provided"
+			logger.Error(ctx, msg)
+			return Device{}, errors.New(msg)
 		}
-		if wwn != "" {
-			for _, d := range devices {
-				if err := c.scsi.WaitUdevSymlinkNVMe(ctx, d, wwn); err == nil {
-					logger.Error(ctx, "registered device found: %s", d)
-					return Device{Name: d, WWN: wwn}, nil
-				}
+		if wwn != "" && nguid != "" {
+			if len(devices) > 1 {
+				logger.Debug(ctx, "Multiple nvme devices found for the given wwn %s", wwn)
```

Comment:
for my understanding, is this scenario due to multipath? is it ok to return the first device alone in line 335?

Reply:
Usually one device is present in the list (Ubuntu and SLES). In case nvme produces multiple devices, we ll log it for reference and use the first device alone. This will be changed once we have a fix from multipath

Reply:
ok.